pilot/pkg/model/authentication_test.go

```diff
@@ -111,3 +116,47 @@ func TestLegacyAuthenticationPolicyToPolicy(t *testing.T) {
 		}
 	}
 }
+
+func TestConstructSdsSecretConfig(t *testing.T) {
+	refreshDelay := 15 * time.Second
+
+	cases := []struct {
+		serviceAccount string
+		meshConfig     *meshconfig.MeshConfig
+		expected       *auth.SdsSecretConfig
+	}{
```

Comment:
In general, tt's nice to have tests for corner/special cases. E.g blank sds path in this case.

Reply:
added more negative test cases.